File: apps/api/app/schemas/comparison.py
import uuid

from pydantic import BaseModel


class CompareRequest(BaseModel):
    document_id_a: uuid.UUID
    document_id_b: uuid.UUID


class ComparisonSideResponse(BaseModel):
    found: bool
    text: str | None
    page: int | None
    section: str | None
    chunk_id: str | None


class ComparisonRowResponse(BaseModel):
    category: str
    label: str
    document_a: ComparisonSideResponse
    document_b: ComparisonSideResponse


class CompareResponse(BaseModel):
    document_id_a: uuid.UUID
    document_id_b: uuid.UUID
    rows: list[ComparisonRowResponse]

File: apps/web/types/comparison.ts
export interface ComparisonSide {
  found: boolean;
  text: string | null;
  page: number | null;
  section: string | null;
  chunk_id: string | null;
}

export interface ComparisonRow {
  category: string;
  label: string;
  document_a: ComparisonSide;
  document_b: ComparisonSide;
}

export interface CompareResponse {
  document_id_a: string;
  document_id_b: string;
  rows: ComparisonRow[];
}
